File: flower.cpp
#include <iostream>
#include <string.h>

using namespace std;


class flower
{
public:
  string color;
  int height;
  float percentWater;

  flower(string newColor, int newHeight, float newPercentWater) {
    color = newColor;
    height = newHeight;
    percentWater = newPercentWater;
  }

  void print()
  {
    cout << color << " (" << height << " cm, "
        << percentWater << "% water)" << endl;
  }

  void grow()
  {
    height += 2;
    percentWater -= 0.05f;
  }

  void water()
  {
    percentWater += 0.1f;
  }
};

int main()
{
  flower f("purple", 10, 0.5f);

  f.print();

  f.water();

  f.print();

  f.grow();

  f.print();

  return 0;
}

File: Flower.java
public class Flower {
  String color;
  int height;
  float percentWater;

  public Flower(String newColor, int newHeight, float newPercentWater) {
    this.color = newColor;
    this.height = newHeight;
    this.percentWater = newPercentWater;
  }

  public void print() {
    System.out.println(" (" + this.height + " cm, " + this.percentWater + "% water)");
  }

  public void grow() {
    this.height += 2;
    this.percentWater -= 0.05f;
  }

  public void water() {
    this.percentWater += 0.1f;
  }

  public static void main(String[] args) {
    Flower f = new Flower("purple", 10, 0.5f);
    f.print();
    f.water();
    f.print();
    f.grow();
    f.print();
  }
}
